samples/bluetooth/fast_pair/locator_tag/include/app_fp_adv.h

```diff
@@ -32,11 +32,51 @@ enum app_fp_adv_mode {
 	APP_FP_ADV_MODE_NOT_DISCOVERABLE,
 };
 
-/** Set the Fast Pair advertising mode.
+/** Fast Pair advertising trigger. */
+struct app_fp_adv_trigger {
+	const char *id;
+	bool enabled;
```

Comment:
Or even better: can we use `void *ctx` instead of this field and assign the memory segments from the static array defined in the fp_adv.c source code?

Reply:
> Or even better: can we use `void *ctx` instead of this field and assign the memory segments from the static array defined in the fp_adv.c source code?

Please see the https://github.com/nrfconnect/sdk-nrf/pull/16529#discussion_r1694874959

We could probably do some calculation of the index based on the trigger struct address and the STRUCT_SECTION_START address and later use it in bitmask in source code to not expose it (then the trigger number would be capped at for example 32 if used uint32_t, but would still require at least to add the ASSERT instead of the define to validate sizes).

However, I think that hiding the implementation is an overkill for the code at the sample level.
We can consider changes in the future if needed (for example if we will decide to move this module to the usecase layer). 
I'll just add the comment as suggested.

Reply:
I think that using sections to implement this is an overkill and made the API more complex. Additionally, we need to expose the trigger state to the external user. Our originally discussed approach with the `int app_fp_adv_trigger_register` (dynamic registration) and the `app_fp_adv_request` API approach was better.

Reply:
See the following simplification:

https://github.com/nrfconnect/sdk-nrf/pull/16529#discussion_r1695159208

Reply:
Why is using section making this code more complex? It seems to significantly simplify trigger registration. It also helps to avoid misconfigurations.

If you want to keep `enable` information private to `app_fp_adv` we could map index in section to bit in bitmask kept internally in `app_fp_adv` (should be straightforward).

Reply:
Changed to bitmask as discussed